hphp/runtime/ext/std/ext_std_process.cpp

```diff
@@ -189,6 +189,8 @@ namespace {
 
 struct ShellExecContext final {
   ShellExecContext() {
+    // Use the default handler while exec'ing in a shell,
+    // saving the previous signal action.
     m_sig_handler = signal(SIGCHLD, SIG_DFL);
```

Comment:
Can you use `sigaction()` here, too?  It makes me a bit woozy to be defining the handler with `sigaction()`, but then resetting and retrieving it with `signal()`.